### PR TITLE
Replace _.min with Math.min

### PR DESCRIPTION
**Pull request by vietduc01100001**
Addresses: #7747

### CODE REVIEW

File: lib/plugins/aws/deploy/lib/checkForChanges.js [hunk 1]
```diff
@@ -98,7 +98,11 @@ module.exports = {
     });
 
     return BbPromise.all(getFunctionResults)
-      .then(_.min)
+      .then(results => {
+        if (!results || results.length === 0) return undefined;
+        const timestamp = Math.min(...results); // Math.min converts Date to timestamp
+        return new Date(timestamp);
```
Comment (medikoo) [inline lib/plugins/aws/deploy/lib/checkForChanges.js:104]: Let's maybe rely on `reduce` instead (?)

```javascript
results.reduce((currentMin, date) => (!currentMin || date < currentMin) ? date : currentMin, undefined)
```

Comment (vietduc01100001) [reply lib/plugins/aws/deploy/lib/checkForChanges.js:104]: I'll try to make this part cleaner.

File: lib/plugins/aws/deploy/lib/checkForChanges.js [hunk 2]
```diff
@@ -98,7 +98,12 @@ module.exports = {
     });
 
     return BbPromise.all(getFunctionResults)
-      .then(_.min)
+      .then(results =>
+        results.reduce((currentMin, date) => {
+          if (!currentMin || date < currentMin) return date;
+          return currentMin;
+        }, undefined)
```
Comment (medikoo) [inline lib/plugins/aws/deploy/lib/checkForChanges.js:105]: We can improve it further, let;s replace `undefined` with `null` and remove `then(..)`  that follows it :)

Comment (vietduc01100001) [reply lib/plugins/aws/deploy/lib/checkForChanges.js:105]: Thanks a lot! I've learned so much from these PR reviews. :D

Comment (medikoo) [reply lib/plugins/aws/deploy/lib/checkForChanges.js]: Great to hear that :)

**Pull request merged**